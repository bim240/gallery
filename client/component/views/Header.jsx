import React from "react";
import { Navbar, Nav, Container } from "react-bootstrap";

const Header = (props) => {
  return (
    <>
      <Navbar className="header_contaienr" collapseOnSelect sticky="top">
        <Container>
          <Navbar.Brand
            title="Home"
            href="/"
            className="font-weight-bold mob_brand text-secondary">
            Gallery
          </Navbar.Brand>

          <Nav className="ml-auto h4_heading mob_nav_bar ">
            <img
              title="Profile Image"
              className="header_image"
              src="https://avatars2.githubusercontent.com/u/50172413?s=400&u=0573967b786828dda99a1efb64dff093fb654f08&v=4"
              alt="profile image"
            />
          </Nav>
        </Container>
      </Navbar>
    </>
  );
};

export default Header;
