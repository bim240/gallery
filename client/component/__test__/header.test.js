import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Header from "../views/Header";

test("render", () => {
  const { container, getByText } = render(<Header />);
  const homeLink = getByText("Gallery");

  fireEvent.click(homeLink);
  expect(homeLink.href).toBe("http://localhost/");
  expect(homeLink.title).toBe("Home");

  expect(container).toMatchSnapshot();
});
