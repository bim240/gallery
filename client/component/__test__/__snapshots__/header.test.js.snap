// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div>
  <nav
    class="header_contaienr navbar navbar-expand navbar-light sticky-top"
  >
    <div
      class="container"
    >
      <a
        class="font-weight-bold mob_brand text-secondary navbar-brand"
        href="/"
        title="Home"
      >
        Gallery
      </a>
      <div
        class="ml-auto h4_heading mob_nav_bar  navbar-nav"
      >
        <img
          alt="profile image"
          class="header_image"
          src="https://avatars2.githubusercontent.com/u/50172413?s=400&u=0573967b786828dda99a1efb64dff093fb654f08&v=4"
          title="Profile Image"
        />
      </div>
    </div>
  </nav>
</div>
`;
